drivers/serial/uart_mcux_flexcomm.c

```diff
@@ -1146,6 +1158,40 @@ static int mcux_flexcomm_init_common(const struct device *dev)
 	return 0;
 }
 
+#if UART_IS_WAKEUP
+static void mcux_flexcomm_pm_prepare_wake(const struct device *dev, enum pm_state state)
```

Comment:
Why do we need PM notifiers. Why are these functions no called from the PM action handler that is implemented in this driver.

Reply:
These notifiers are invoked during system power transitions as well. So it seems like we have something already available in the driver. 
We can move these calls to the SUSPEND and RESUME actions.

Reply:
I am not sure I follow. The place that invokes the notifiers in the PM code is where the driver action handlers are also invoked. Why can't we call these functions from the action handler, maybe we can place these outside the switch case statement in the action handler.

Reply:
You are welcome to pull down the PR and try to make whatever changes to prove us wrong, I cannot act on your comments because they don't make sense to me